cloudant-client/src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -114,8 +114,19 @@ public void cookieTest() {
      * Assert that the User-Agent header is of the expected form.
      */
     @Test
-    public void testUserAgentHeaderString() {
-        String userAgentHeader = new LibraryVersion().getUserAgentString();
+    public void testUserAgentHeaderString() throws Exception {
+
+        ClientBuilder builder = ClientBuilder.account("test");
+        Class clazz = builder.getClass();
+        Field interceptor = clazz.getDeclaredField("USER_AGENT_INTERCEPTOR");

```

Comment:
I think if we move the logic into the `UserAgentInterceptor` we could new up using the prefix string and avoid all the reflection.